File: Professional/ActiveReportsWithMVC5WebAPI/VB.NET/ActiveReportsWithMVC5/Models/CustomerContext.vb
Imports System
Imports System.Collections.Generic
Imports System.Configuration
Imports System.Data.OleDb
Public Class CustomerContext
	Public Shared Customers As List(Of Customer)
	Public Shared customer As Customer
	''' <summary>
	''' Gets customers from CustomerOrders table of Reels.mdb
	''' <returns>list of Customers</returns>
	''' </summary>
	Public Shared Function GetCustomers() As List(Of Customer)
		If Customers Is Nothing Then
			Dim allcustomers As New List(Of Customer)
			Dim conn As New OleDbConnection(ConnectString())
			conn.Open()
			Dim command As New OleDbCommand("select Distinct Top 30 CustomerID, FirstName,LastName,MiddleInitial from CustomerOrders", conn)
			Dim dr As OleDbDataReader = command.ExecuteReader()
			While dr IsNot Nothing AndAlso dr.Read
				allcustomers.Add(New Customer With _
				{
					.ContactName = dr("FirstName").ToString() & " " & dr("MiddleInitial").ToString() & " " & dr("LastName").ToString(),
					.CustomerId = Convert.ToInt32(dr("CustomerID").ToString())
				})
			End While
			conn.Close()
			Customers = allcustomers
		End If
		Return Customers
	End Function
	''' <summary>
	''' Reads the ConnectionString for the Reels.mdb from the web.config file 
	''' <returns>connectionstring:-"ConnectionString" from web.config</returns>
	''' </summary>
	Private Shared Function ConnectString() As String
		Return ConfigurationManager.ConnectionStrings("ConnectionString").ConnectionString
	End Function
	''' <summary>
	''' Gets customerdetail from CustomerOrders table of based on CustomerID 
	''' <returns>Customer object</returns>
	''' </summary>
	Friend Shared Function GetCustomersDetail(id As Integer) As Customer
		Dim conn As New OleDbConnection(ConnectString())
		conn.Open()
		Dim command As New OleDbCommand("Select * from CustomerOrders where CustomerID=" & id, conn)
		Dim dr As OleDbDataReader = command.ExecuteReader()
		While dr IsNot Nothing AndAlso dr.Read
			customer = New Customer With _
			{
				.Address = dr("Address1").ToString(),
				.City = dr("City").ToString(),
				.ContactName = dr("FirstName").ToString() & " " & dr("MiddleInitial").ToString() & " " & dr("LastName").ToString(),
				.CustomerId = Convert.ToInt32(dr("CustomerID").ToString()),
				.Country = dr("Country").ToString(),
				.PostalCode = dr("PostalCode").ToString(),
				.Region = dr("Region").ToString()
			}
		End While
		conn.Close()
		Return customer
	End Function
End Class

File: Professional/ActiveReportsWithMVC5WebAPI/C#/ActiveReportsWithMVC5/Models/CustomerContext.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data.OleDb;
namespace ActiveReportsWithMVC5.Models
{
	public class CustomerContext 
	{
		public static List<Customer> Customers;
		public static Customer customer;	   
		/// <summary>
		/// Gets customers from CustomerOrders table of Reels.mdb
		/// <returns>list of Customers</returns>
		/// </summary>
		public static List<Customer> GetCustomers()
		{
			if (Customers == null)
			{
				var allcustomers = new List<Customer>();
				var conn = new OleDbConnection(ConnectString());
				conn.Open();
				var command = new OleDbCommand("select Distinct Top 30 CustomerID, FirstName,LastName,MiddleInitial from CustomerOrders", conn);
				var dr = command.ExecuteReader();
				while (dr != null && dr.Read())
				{
					allcustomers.Add(new Customer
					{
						ContactName = dr["FirstName"].ToString() +" "+dr["MiddleInitial"].ToString()+" "+ dr["LastName"].ToString(),						
						CustomerId = Convert.ToInt32(dr["CustomerID"].ToString())
					});
				}
				conn.Close();
				Customers = allcustomers;
			}
			return Customers;
		}
		/// <summary>
		/// Reads the ConnectionString for the Reels.mdb from the web.config file	  
		/// <returns>connectionstring:-"ConnectionString" from web.config</returns>
		/// </summary>
		private static string ConnectString()
		{
			return ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
		}
		/// <summary>
		/// Gets customerdetail from CustomerOrders table of based on CustomerID	   
		/// <returns>Customer object顧客オブジェクト</returns>
		/// </summary>
		internal static Customer GetCustomersDetail( int id)
		{
				var conn = new OleDbConnection(ConnectString());
				conn.Open();
				var command = new OleDbCommand("Select * from CustomerOrders where CustomerID="+id, conn);
				var dr = command.ExecuteReader();
				while (dr != null && dr.Read())
				{
					customer=new Customer
					{
						Address = dr["Address1"].ToString(),
						City = dr["City"].ToString(),					   
						ContactName = dr["FirstName"].ToString() +" "+dr["MiddleInitial"].ToString()+" "+ dr["LastName"].ToString(),						
						CustomerId = Convert.ToInt32(dr["CustomerID"].ToString()),						
						Country = dr["Country"].ToString(),										   
						PostalCode = dr["PostalCode"].ToString(),
						Region = dr["Region"].ToString()
					};
				}
				conn.Close();
			return customer;
		}
	}
}
